### ISSUE TITLE
将Boundary Smoothing应用于其他模型的问题

### ISSUE THREAD

**Issue by happyZew** [syuoni/eznlp#49]
老师您好，想将Boundary Smoothing应用于其他模型，模型计算交叉熵时traget是一维张量代表若干个span的类别，Boundary Smoothing得到的label_ids怎么用于计算loss呢，请问老师这部分在源码的哪里，恳请老师解惑，不胜感激。

**Comment by happyZew** [comment]
```
def soft_label_cross_entropy(logits: torch.Tensor, soft_target: torch.Tensor, weight: torch.Tensor=None, reduction: str='none'):
    """Soft label cross entropy loss.
    
    Parameters
    ----------
    logits : torch.Tensor (num_entries, logit_dim)
        Logits before softmax. 
    soft_target : torch.Tensor (num_entries, logit_dim)
        The ground-truth distribution over indexes, s.t. `target.sum(dim=-1)` equals 1 in all entries.
    weight : torch.Tensor (logit_dim, )
        A manual rescaling weight given to each class. 
    """
    _check_soft_target(soft_target)
    
    log_prob = logits.log_softmax(dim=-1)
    
    if weight is not None:
        log_prob = log_prob * weight
    
    losses = -(log_prob * soft_target).sum(dim=-1)
    return _reduce_losses(losses, sample_weight=None, reduction=reduction)
```
没问题了老师，在这里找见计算过程了，谢谢老师

**Comment by syuoni** [comment]
Boundary Smoothing 的 label_ids 是在实体类别上的一个 概率分布。
Loss 计算是 soft cross entropy loss：https://github.com/syuoni/eznlp/blob/master/eznlp/nn/functional.py#L189-L209

**Comment by happyZew** [comment]
非常感谢您的回复，最先我想自己写函数去计算交叉熵，在您的源码中也找到了这一函数，后来我发现可以使用CrossEntropyLoss去做同样shape的两个二维张量的交叉熵，解决了这个问题，于是关闭了issue，再次感谢您的回复

**Comment by Huyangjuan** [comment]
同学您好，请问您有没有成功将Boundary Smoothing应用于你的模型呢，可以请教一下你是加了哪些地方的哪些代码嘛？非常期待您的回复，或者您可以直接跟我交流，这是我的QQ：206859629

**Comment by Huyangjuan** [comment]
> 非常感谢您的回复，最先我想自己写函数去计算交叉熵，在您的源码中也找到了这一函数，后来我发现可以使用CrossEntropyLoss去做同样shape的两个二维张量的交叉熵，解决了这个问题，于是关闭了issue，再次感谢您的回复

您好，请问可以请教一下您是如何使用损失函数来计算这个边界平滑是损失嘛？您是使用了什么损失函数代码呢？非常感谢您的回复！
